nomad/core_sched.go

```diff
@@ -986,18 +1020,23 @@ func (c *CoreScheduler) batchRotateVariables(iter memdb.ResultIterator, eval *st
 		if err != nil {
 			return err
 		}
-		variables = append(variables, dv)
-		if len(variables) == 20 {
-			err := upsertFn(variables)
-			if err != nil {
-				return err
-			}
-			variables = []*structs.SecureVariableDecrypted{}
+		args.Var = dv
+		reply := &structs.SecureVariablesApplyResponse{}
```

Comment:
I've been thinking about hashes over the content to determine if the content has changed earlier in the process. 
Since rekeying uses the Apply RPC, rekeys are going to always appear to be meaningful writes to blocking queries.  As far as the user is concerned they're noops, so is that desirable? For my edification, could using the original ModifyIndex present in the original item metadata prevent that scenario?

Reply:
> Since rekeying uses the Apply RPC, rekeys are going to always appear to be meaningful writes to blocking queries. As far as the user is concerned they're noops, so is that desirable? For my edification, could using the original ModifyIndex present in the original item metadata prevent that scenario?

It's not ideal for sure. Consumers of those blocking queries should ideally be smart about detecting diffs in the API output (the way that consul-template is about its rendered outputs) to avoid triggering downstream events unnecessarily. Not changing the `ModifyIndex` won't help with blocking queries, as blocking queries will be waiting on the table index.

(Vault gets away with this by not supporting blocking queries! But that's not very nice.)